docs/flask.md

```diff
@@ -0,0 +1,202 @@
+the Flask server run configuration takes care of this for you, and you don't
+need the startup helper script. The module it should use is `wsgi`.
+
+The run script supports setting environment variables PYTHON and COMMAND to
+choose your Python interpreter. By default, it runs `python3 -m flask`, but you
+can set `COMMAND=flask` to have it call `flask` directly, or `PYTHON=python3.7`
+to have it run `python3.7 -m flask`, for example.
+
+## Adding or updating dependencies
+
+Stager uses [pip-tools](https://github.com/jazzband/pip-tools) to manage
+`requirements.txt` for production and `requirements-dev.txt` in development.
+It needs to be installed separately and is not included in the Docker dev image
+for now.
+
+To add a runtime dependency, add a line specifying the name of the PyPI package
+to `requirements.in`. Keep in mind to run these in a virtualenv if you are using
+one!
+
+```bash
+pip-compile requirements.in
+pip install -r requirements.txt
+```
+
+To add a development dependency, add the line to `requirements-dev.in` instead.
+
+```bash
+pip-compile requirements-dev.in
+pip install -r requirements-dev.in
+```
+
+Commit the changes to Git. If you are using Docker, you should rebuild the image
+for changes to persist.
+
+## Application architecture
+
+The backend is a RESTful API only. It is not responsible for serving the frontend or HTTPS.
+The main entrypoint is [`wsgi.py`](https://github.com/ccmbioinfo/stager/blob/master/flask/wsgi.py),
+which instructs Flask (or any WSGI server in [production](https://github.com/ccmbioinfo/stager/blob/master/docs/production.md))
+to create an app instance with the default config and start the server to listen for HTTP requests.
+
+Configuration is mostly read from environment variables in [`config.py`](https://github.com/ccmbioinfo/stager/blob/master/flask/app/config.py).
+Because [Flask-Login](https://flask-login.readthedocs.io/) is authenticates via a session cookie by default,
+in development, it is convenient to add `LOGIN_DISABLED = True` to the end of the config object for
+testing your endpoints. Most routes are configured to assume that you are an admin if logins are
+disabled, with an option to assume a specific user identity with `user=PRIMARY_KEY` in the request
+query string.
+
+The app instance is assembled in a factory function in [`__init__.py`](https://github.com/ccmbioinfo/stager/blob/master/flask/app/__init__.py).
+Each subroute is loaded via a Flask [Blueprint](https://flask.palletsprojects.com/en/1.1.x/tutorial/views/).
+This also registers the SQLAlchemy ORM and Flask-Migrate. More on this in the
+[database docs](https://github.com/ccmbioinfo/stager/blob/master/docs/database.md).
+
+The `add-default-admin` and `add-default-data` commands are registered to aid in development.
+They are automatically run by the startup shell script. Most of the remaining files each
+hold a Blueprint and implement a set of endpoints for that subroute.
+
+## Tips
+
```

Comment:
would it be worth the time to add a small section on how to add new click (command line) commands and extensions or are these set in stone at this point of development?

Reply:
The commands aren't set in stone but I think the best I can do is link to relevant documentation instead of reproducing it here.